cfpb/consumerfinance.gov
# Only check live EventPages for archiving; append date to duplicate slugs

Drafts that got archived were causing errors when, later, pages with the same slug were attempted to be archived again. This update only checks for live pages to archive (and only EventPages now), and if it finds a duplicate slug, it will append the event's start date to the slug before moving it.


## How to test this PR

1. Refresh your DB with last night's dump
1. Pull branch
1. Run `./cfgov/manage.py runscript archive_events`
1. Verify that it returns `No past events to be archived were found.`
1. Publish the draft current events with dates in the past
1. Run `./cfgov/manage.py runscript archive_events` again
1. Verify that it moves them to the archive, with their start date appended to the slug

## Notes and todos

- [x] Convert to management command
- [ ] Update Jenkins job to use the new management command


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
